src/backend/electron/ipc.ts

```diff
@@ -1,45 +1,69 @@
-import { ipcMain, IpcMainInvokeEvent, BrowserWindow } from "electron";
+import {
+  BrowserWindow,
+  ipcMain,
+  IpcMainInvokeEvent,
+  IpcRendererEvent,
+} from "electron";
 import log from "electron-log/main";
 import { IpcIHData, IpcSOData } from "@/type/ipc";
 
-export function ipcMainHandle<T extends keyof IpcIHData>(
-  channel: T,
-  listener: (
-    event: IpcMainInvokeEvent,
-    ...args: IpcIHData[T]["args"]
-  ) => IpcIHData[T]["return"] | Promise<IpcIHData[T]["return"]>,
-): void;
-export function ipcMainHandle(
-  channel: string,
-  listener: (event: IpcMainInvokeEvent, ...args: unknown[]) => unknown,
-): void {
-  const errorHandledListener = (
+export type IpcRendererInvoke = {
+  [K in keyof IpcIHData]: (
+    ...args: IpcIHData[K]["args"]
+  ) => Promise<IpcIHData[K]["return"]>;
+};
+
+export type IpcMainHandle = {
+  [K in keyof IpcIHData]: (
     event: IpcMainInvokeEvent,
-    ...args: unknown[]
-  ) => {
-    try {
-      validateIpcSender(event);
-      return listener(event, ...args);
-    } catch (e) {
-      log.error(e);
-    }
-  };
-  ipcMain.handle(channel, errorHandledListener);
-}
+    ...args: IpcIHData[K]["args"]
+  ) => Promise<IpcIHData[K]["return"]> | IpcIHData[K]["return"];
+};
 
-export function ipcMainSend<T extends keyof IpcSOData>(
-  win: BrowserWindow,
-  channel: T,
-  ...args: IpcSOData[T]["args"]
+export type IpcMainSend = {
+  [K in keyof IpcSOData]: (
+    win: BrowserWindow,
+    ...args: IpcSOData[K]["args"]
+  ) => void;
+};
+
+export type IpcRendererOn = {
+  [K in keyof IpcSOData]: (
+    event: IpcRendererEvent,
+    ...args: IpcSOData[K]["args"]
+  ) => Promise<IpcSOData[K]["return"]> | IpcSOData[K]["return"];
+};
+
+// FIXME: asをなるべく使わずに書いたらこうなった。もっとキレイな書き方があれば…
+export function registerIpcMainHandle<T extends IpcMainHandle>(
```

Comment:
（細かいのですが）
コメント内で指示語はなるべく使わないようにするのが良いとされてたりします！
曖昧になっちゃうので。

「こう」「もっとキレイ」を展開すると、こう･･･？
```suggestion
// FIXME: asを使わないようオーバーロードにした。オーバーロードも使わない書き方にしたい。
export function registerIpcMainHandle<T extends IpcMainHandle>(
```